src/cosmo.jl

```diff
@@ -0,0 +1,151 @@
+"""
+    AbstractCosmology{T}
+An abstract type representing a general cosmological model.
+"""
+abstract type AbstractCosmology{T} end
+
+"""
+    AbstractCosmologicalGrid{T}
+An abstract type representing a grid on which cosmological quantities are evaluated.
+"""
+abstract type AbstractCosmologicalGrid{T} end
+
+"""
+    AbstractBackgroundQuantities{T}
+An abstract type for background quantities in cosmology, such as the Hubble parameter (`H`), comoving distance (`χ`), 
+and the growth factor (`D`).
+"""
+abstract type AbstractBackgroundQuantities{T} end
+
+"""
+    AbstractCosmologicalProbes{T}
+An abstract type for the shear, clustering and CMB lensing kernels.
+"""
+abstract type AbstractCosmologicalProbes{T} end
+
+
+# Define the flat ΛCDM cosmological model with default parameters based on the fiducial N5K cosmology.
+
+"""
+    FlatΛCDM{T}(; w0 = -1.0, wa = 0.0, H0 = 67.27, Ωm = 0.3156, Ωb = 0.0492, 
+        Ωde = 0.6844, As = 2.12107e-9, σ8 = 0.816, Ωk = 0.0, Ωr = 0.0, ns = 0.9645
+    )
+
+# Parameters:
+- `w0`: Dark energy equation of state parameter at present time (default: -1).
+- `wa`: Time evolution of the dark energy equation of state (default: 0).
+- `H0`: Hubble constant in km/s/Mpc (default: 67.27).
+- `Ωm`: Matter density parameter (default: 0.3156).
+- `Ωb`: Baryon density parameter (default: 0.0492).
+- `Ωde`: Dark energy density parameter (default: 0.6844).
+- `As`: Scalar amplitude of the primordial power spectrum (default: 2.12107e-9).
+- `σ8`: Root-mean-square density fluctuation in spheres of radius 8 Mpc (default: 0.816).
+- `Ωk`: Curvature density parameter (default: 0, for flat universe).
+- `Ωr`: Radiation density parameter (default: 0, since radiation is negligible at low redshift).
+- `ns`: Scalar spectral index (default: 0.9645).
+"""
+@kwdef mutable struct FlatΛCDM{T} <: AbstractCosmology{T}
+    #TODO: comology will need updates, A_s and sigma8 are not independent of eachother
+    w0::T  = -1.0
+    wa::T  = 0.0
+    H0::T  = 67.27
+    Ωm::T  = 0.3156
+    Ωb::T  = 0.0492
+    Ωde::T = 0.6844
+    As::T  = 2.12107e-9
+    σ8::T  = 0.816
+    Ωk::T  = 0.0
+    Ωr::T  = 0.0
+    ns::T  = 0.9645
+end
```

Comment:
There is something I do not like. The struct is `FlatLCDM`, but still allows for `w0` and `wa`.